contracts/router/modules/bridge/SynapseBridgeModule.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {OnlyDelegateCall} from "../pool/OnlyDelegateCall.sol";
+import {IBridgeModule} from "../../interfaces/IBridgeModule.sol";
+import {ILocalBridgeConfig} from "../../interfaces/ILocalBridgeConfig.sol";
+import {ISynapseBridge} from "../../interfaces/ISynapseBridge.sol";
+import {Action, BridgeToken, DefaultParams, SwapQuery} from "../../libs/Structs.sol";
+import {UniversalTokenLib} from "../../libs/UniversalToken.sol";
+
+contract SynapseBridgeModule is OnlyDelegateCall, IBridgeModule {
+    using UniversalTokenLib for address;
+
+    error SynapseBridgeModule__EqualSwapIndexes(uint8 index);
+    error SynapseBridgeModule__UnsupportedDepositAction(Action action);
+    error SynapseBridgeModule__UnsupportedRedeemAction(Action action);
+    error SynapseBridgeModule__UnsupportedToken(address token);
+
+    /// These need to be immutable in order to be accessed via delegatecall
+    ILocalBridgeConfig public immutable localBridgeConfig;
+    ISynapseBridge public immutable synapseBridge;
+
+    constructor(address localBridgeConfig_, address synapseBridge_) {
+        localBridgeConfig = ILocalBridgeConfig(localBridgeConfig_);
+        synapseBridge = ISynapseBridge(synapseBridge_);
+    }
+
+    /// @inheritdoc IBridgeModule
+    function delegateBridge(
+        address to,
+        uint256 chainId,
+        address token,
+        uint256 amount,
+        SwapQuery memory destQuery
+    ) external payable {
```

Comment:
Any thoughts on removing `payable` given the delegate call. Unclear we need it and reminds me of this issue every time I see it https://github.com/Uniswap/v3-periphery/issues/52 (although not the same as multicall here)